### ISSUE TITLE
testcamera does not work on emscripten

### ISSUE THREAD

**Issue by madebr** [libsdl-org/SDL#10563]
This is printed in the console, and the window remains grey:
```
GET
http://localhost:8080/favicon.ico
[HTTP/1 404 File not found 2ms]

emscripten_set_main_loop_timing: Cannot set timing mode for main loop since a main loop does not exist! Call emscripten_set_main_loop first to set one up. testcamera.js:4211:12
Saw 1 camera devices. testcamera.js:1565:16
  - Camera #0:  Web browser's camera testcamera.js:1565:16
Available formats: testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera permission has not been granted testcamera.js:1565:16
Camera is opened! Actual spec: (640x480), fps=30 testcamera.js:994:1262
Camera approved! testcamera.js:1565:16
Camera Spec: 640x480 30.00 FPS SDL_PIXELFORMAT_UNKNOWN testcamera.js:1565:16
Parameter 'w' is invalid testcamera.js:1565:16
Texture dimensions can't be 0 testcamera.js:1565:16
Couldn't create texture: Texture dimensions can't be 0 testcamera.js:1565:16
Uncaught ReferenceError: _SDL_free is not defined
    181841 http://localhost:8080/testcamera.js:995
    runMainThreadEmAsm http://localhost:8080/testcamera.js:4195
    _emscripten_asm_const_int_sync_on_main_thread http://localhost:8080/testcamera.js:4197
    callUserCallback http://localhost:8080/testcamera.js:4460
    runIter http://localhost:8080/testcamera.js:4535
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4421
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
    requestAnimationFrame http://localhost:8080/testcamera.js:4830
    Browser_mainLoop_scheduler_rAF http://localhost:8080/testcamera.js:4227
    Browser_mainLoop_runner http://localhost:8080/testcamera.js:4434
testcamera.js:995:247
That operation is not supported 4 testcamera.js:1565:16
```

Chrome points at this location:
https://github.com/libsdl-org/SDL/blob/a7bed810b3499d157655ae6dcc7cd5e8a34fa549/src/camera/emscripten/SDL_camera_emscripten.c#L87-L95

**Comment by icculus** [comment]
Hmm, there's some magic to tell Emscripten "this function needs to exist and be findable under a real symbol from JavaScript."

I'll dig that up.

**Comment by icculus** [comment]
So it doesn't look like this field is actually used, so removing the _SDL_free() line was the solution.

This (and other Emscripten fixes) are in 405693175db7d5c076588f4cce4e1add48d8e9dd.
